cmd/state-svc/internal/resolver/resolver.go

```diff
@@ -178,9 +173,9 @@ func (r *Resolver) RuntimeUsage(ctx context.Context, pid int, exec string, dimen
 func (r *Resolver) CheckDeprecation(ctx context.Context) (*graph.DeprecationInfo, error) {
 	logging.Debug("Check deprecation resolver")
 
-	deprecated, err := r.deprecation.Check()
-	if err != nil {
-		return nil, errs.Wrap(err, "Could not fetch deprecation information")
+	deprecated, ok := r.depPoller.ValueFromCache().(*graph.DeprecationInfo)
```

Comment:
This won't panic if the value from `ValueFromCache()` is nil right? I'm pretty sure it won't. Might be something nice to confirm in the unit test for the `poller`

Reply:
I can't confirm it in the poller, because the value type is an interface and we can't reliably assert nil unless we know the type.

We're not doing anything with the returned value here, other than also return it. And as per the contract of the function it CAN return a nil, which isn't a new change.